lib/rules/quote-props.js

```diff
@@ -141,6 +141,8 @@ module.exports = function(context) {
 
                     necessaryQuotes = necessaryQuotes || !areQuotesRedundant(tokens) || KEYWORDS && isKeyword(tokens[0].value);
                 }
+            } else if (KEYWORDS && checkQuotesRedundancy && key.type === "Identifier" && isKeyword(key.name)) {
+                context.report(node, "Properties should be quoted as `{{property}}` is a reserved word.");
```

Comment:
I think you should report the property node to be consistent with rule's other warnings around this.